packages/scripts/src/admin-proposals/changeFinderAddress.js

```
@@ -0,0 +1,273 @@
+    console.log(`- Registry @ ${contractsByNetId[chainId].registry.options.address}`);
+    console.log(
+      `- ${chainId === 137 ? "GovernorRootTunnel" : "GovernorHub"} @ ${
+        contractsByNetId[chainId].l1Governor.options.address
+      }`
+    );
+    console.groupEnd();
+  }
+
+  const gasEstimator = await setupGasEstimator();
+
+  if (!verify) {
+    console.group("\n🌠 Proposing new Admin Proposal");
+
+    const adminProposalTransactions = [];
+    console.group("\n Key to understand the following logs:");
+    console.log(
+      "- 🟣 = Transactions to be submitted to the Polygon contracts are relayed via the GovernorRootTunnel on Ethereum. Look at this test for an example:"
+    );
+    console.log(
+      "    - https://github.com/UMAprotocol/protocol/blob/349401a869e89f9b5583d34c1f282407dca021ac/packages/core/test/polygon/e2e.js#L221"
+    );
+    console.log(
+      "- 🔴 = Transactions to be submitted to networks with GovernorSpokes contracts are relayed via the GovernorHub on Ethereum. Look at this test for an example:"
+    );
+    console.log(
+      "    - https://github.com/UMAprotocol/protocol/blob/0d3cf208eaf390198400f6d69193885f45c1e90c/packages/core/test/cross-chain-oracle/chain-adapters/Arbitrum_ParentMessenger.js#L253"
+    );
+    console.log("- 🟢 = Transactions to be submitted directly to Ethereum contracts.");
+    console.groupEnd();
+
+    if (ethereum) {
+      console.group(`\n🟢 Pointing Finder for "${contract}" to ${ethereum}`);
+      let skip;
+      try {
+        const implementationAddress = await mainnetContracts.finder.methods
+          .getImplementationAddress(utf8ToHex(contract))
+          .call();
+        if (implementationAddress === toChecksumAddress(ethereum)) {
+          skip = true;
+          console.log("- Contract @ ", ethereum, `is already set to ${contract} in the Finder. Nothing to do.`);
+        }
+      } catch (err) {
+        // eslint-disable-line prettier/prettier
```

Why do we ignore errors?

---

Added a comment:

```
        // If `getImplementationAddress` reverts, then implementation is not set yet so we shouldn't skip this
        // proposal. Not having an implementation set is an expected situation for this script so we should ignore this
        // error.
```